### PR TITLE
hf dataset added

### PR DESCRIPTION
**Pull request by anshumanpatil**
Completed : - 

**New dataset added related to fitness**

- Changed dataset formatting and created pre training data.
 
- Used dataset - [fitness-question-answers](https://huggingface.co/datasets/its-myrto/fitness-question-answers) 
 
- Created pre training data with - [ibm/granite-4-h-tiny](https://huggingface.co/ibm-granite/granite-4.0-h-tiny)
 
Tried but unsuccessful : -

**Working on** 

Adjusting hyperparameters.

- Changed `MAX_SEQ_LEN` and found that 180 works.

- Still working on `learning rate` and `batch size`

Any Suggestion/Help is welcome 
Thanks in advance

### CODE REVIEW

**Comment by tekaratzas** [comment]
Very cool! How is the training time with this?
